src/types.d.ts

```diff
@@ -0,0 +1,148 @@
+type Color = [number, number, number, number] | [number, number, number];
+
+type Category = 'category' | 'value1' | 'valueA' | 'valueZ' | 'z';
+type Value = 'value' | 'value2' | 'valueB' | 'valueW' | 'w';
+type DataEncoding = Category | Value;
+type PointDataEncoding = DataEncoding | 'inherit' | 'segment';
+
+type KeyAction = 'lasso' | 'rotate' | 'merge';
+type KeyMap = Record<'alt' | 'cmd' | 'ctrl' | 'meta' | 'shift', KeyAction>;
+
+type MouseMode = 'panZoom' | 'lasso' | 'rotate';
+
+type Camera2D = ReturnType<import('camera-2d-simple')>;
+type Scale = import('d3-scale').ScaleContinuousNumeric<number, number>;
+
+type Nullifiable = {
+  backgroundImage: null | import('regl').Texture2D | string;
+  colorBy: null | DataEncoding;
+  sizeBy: null | DataEncoding;
+  opacityBy: null | DataEncoding;
+  xScale: null | Scale;
+  yScale: null | Scale;
+  pointConnectionColorBy: null | PointDataEncoding;
+  pointConnectionOpacityBy: null | PointDataEncoding;
+  pointConnectionSizeBy: null | PointDataEncoding;
+};
+
+type PointOptions = {
+  pointColor: Color;
+  pointColorActive: Color;
+  pointColorHover: Color;
+  pointOutlineWidth: number;
+  pointSize: number;
+  pointSizeSelected: number;
+  pointConnectionColor: Color;
+  pointConnectionColorActive: Color;
+  pointConnectionColorHover: Color;
+  pointConnectionOpacity: number;
+  pointConnectionOpacityActive: number;
+  pointConnectionSize: number;
+  pointConnectionSizeActive: number;
+  pointConnectionMaxIntPointsPerSegment: number;
+  pointConnectionTolerance: number;
+};
+
+type CameraOptions = {
+  cameraTarget: [number, number];
+  cameraDistance: number;
+  cameraRotation: number;
+  cameraView: Float32Array;
+};
+
+type LassoOptions = {
+  lassoColor: Color;
+  lassoLineWidth: number;
+  lassoMinDelay: number;
+  lassoMinDist: number;
+  lassoClearEvent: 'lassoEnd' | 'deselect';
+  lassoInitiator: boolean;
+  lassoInitiatorParentElement: HTMLElement;
+};
+
+export type Settable = {
+  backgroundColor: Color;
+  deselectOnDblClick: boolean;
+  deselectOnEscape: boolean;
+  keyMap: KeyMap;
+  mouseMode: MouseMode;
+  showPointConnections: boolean;
+  showReticle: boolean;
+  reticleColor: Color;
+  opacity: number;
+  opacityByDensityFill: number;
+  height: 'auto' | number;
+  width: 'auto' | number;
+  gamma: number;
+} & PointOptions &
+  CameraOptions &
+  LassoOptions &
+  Nullifiable;
+
+export type Properties = {
+  canvas: HTMLCanvasElement;
+  regl: import('regl').Regl;
+  syncEvents: boolean;
+  version: string;
+  lassoInitiatorElement: HTMLElement;
+  camera: Camera2D;
+  performanceMode: boolean;
+  opacityByDensityDebounceTime: number;
+} & Settable;
```

Comment:
When you call `plot.get` you can return any of the properties on this type, including Settable properties.